src/Compilers/Core/MSBuildTaskTests/GenerateMSBuildAnalyzerConfigTests.cs

```diff
@@ -0,0 +1,303 @@
+
+            GenerateMSBuildAnalyzerConfig configTask = new GenerateMSBuildAnalyzerConfig()
+            {
+                MetadataItems = new[] { item1, item2, item3 }
+            };
+            configTask.Execute();
+
+            var result = configTask.ConfigFileContents;
+
+            Assert.Equal(@"is_global = true
+
+[c:/file1.cs]
+msbuild_item.Compile.ToRetrieve = abc123
+
+[c:/file2.cs]
+msbuild_item.Compile.ToRetrieve = def456
+
+[c:/file3.cs]
+msbuild_item.AdditionalFiles.ToRetrieve = ghi789
+", result);
+        }
+
+        [Fact]
+        public void DuplicateItemSpecsAreCombinedInSections()
+        {
+            TaskItem item1 = new TaskItem("c:\\file1.cs", new Dictionary<string, string> { { "ItemType", "Compile" }, { "MetadataName", "ToRetrieve" }, { "ToRetrieve", "abc123" } });
+            TaskItem item2 = new TaskItem("c:\\file1.cs", new Dictionary<string, string> { { "ItemType", "AdditionalFile" }, { "MetadataName", "ToRetrieve" }, { "ToRetrieve", "def456" } });
+
+            GenerateMSBuildAnalyzerConfig configTask = new GenerateMSBuildAnalyzerConfig()
+            {
+                MetadataItems = new[] { item1, item2 }
+            };
+            configTask.Execute();
+
+            var result = configTask.ConfigFileContents;
+
+            Assert.Equal(@"is_global = true
+
+[c:/file1.cs]
+msbuild_item.Compile.ToRetrieve = abc123
+msbuild_item.AdditionalFile.ToRetrieve = def456
+", result);
+        }
+
+        [Fact]
+        public void ItemIsMissingRequestedMetadata()
+        {
+            TaskItem item1 = new TaskItem("c:\\file1.cs", new Dictionary<string, string> { { "ItemType", "Compile" }, { "MetadataName", "ToRetrieve" } });
+
+            GenerateMSBuildAnalyzerConfig configTask = new GenerateMSBuildAnalyzerConfig()
+            {
+                MetadataItems = new[] { item1 }
+            };
+            configTask.Execute();
+
+            var result = configTask.ConfigFileContents;
+
+            Assert.Equal(@"is_global = true
+
+[c:/file1.cs]
+msbuild_item.Compile.ToRetrieve = 
+", result);
+        }
+
+        [Fact]
+        public void ItemIsMissingRequiredMetadata()
+        {
+            TaskItem item1 = new TaskItem("c:\\file1.cs", new Dictionary<string, string> { });
+            TaskItem item2 = new TaskItem("c:\\file1.cs", new Dictionary<string, string> { { "ItemType", "Compile" } });
+            TaskItem item3 = new TaskItem("c:\\file1.cs", new Dictionary<string, string> { { "MetadataName", "ToRetrieve" } });
+
+            GenerateMSBuildAnalyzerConfig configTask = new GenerateMSBuildAnalyzerConfig()
+            {
+                MetadataItems = new[] { item1, item2, item3 }
+            };
+            configTask.Execute();
+
+            var result = configTask.ConfigFileContents;
+
+            Assert.Equal(@"is_global = true
+
+[c:/file1.cs]
+msbuild_item.. = 
```

Comment:
Is this legal editor config?

Reply:
Yes. It's just a string. It's semantically uninteresting to us, but valid.

---
In reply to: [419847799](https://github.com/dotnet/roslyn/pull/43617#discussion_r419847799) [](ancestors = 419847799)